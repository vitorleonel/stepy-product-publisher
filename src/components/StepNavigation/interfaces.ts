export interface IStepNavigationProps {
  prevText?: string;
  nextText?: string;
  nextIcon?: string;
  nextDisabled?: boolean;
  prevHandler: () => void;
  nextHandler: () => void;
}
