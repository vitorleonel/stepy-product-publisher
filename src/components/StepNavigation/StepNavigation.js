import Button from '../Button';

const StepNavigation = ({
  prevText,
  nextText,
  nextIcon,
  nextDisabled,
  prevHandler,
  nextHandler,
}) => {
  return (
    <div className="steps-item__navigation">
      <Button
        leftIcon="arrow-back-outline"
        text={prevText || 'Previous'}
        type="secondary"
        onClick={prevHandler}
        data-testid="prevButton"
      />

      <Button
        rightIcon={nextIcon || 'arrow-forward-outline'}
        text={nextText || 'Next'}
        type="primary"
        disabled={nextDisabled}
        onClick={nextHandler}
        data-testid="nextButton"
      />
    </div>
  );
};

export default StepNavigation;
