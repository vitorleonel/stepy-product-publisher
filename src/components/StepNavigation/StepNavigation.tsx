import { Button } from "@blueprintjs/core";

import { ViewProps } from "./interfaces";

const StepNavigation = ({
  prevText,
  nextText,
  nextIcon,
  nextLoading,
  nextDisabled,
  prevHandler,
  nextHandler,
}: ViewProps) => {
  return (
    <div className="steps-item__navigation">
      <Button
        icon="arrow-left"
        text={prevText || 'Previous'}
        large
        onClick={prevHandler}
        data-testid="prevButton"
      />

      <Button
        rightIcon={nextIcon || 'arrow-right'}
        text={nextText || 'Next'}
        large
        intent="success"
        loading={nextLoading}
        disabled={nextDisabled}
        onClick={nextHandler}
        data-testid="nextButton"
      />
    </div>
  );
}

export default StepNavigation;
