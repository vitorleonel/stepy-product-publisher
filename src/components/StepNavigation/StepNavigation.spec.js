import { render, fireEvent } from '@testing-library/react';

import StepNavigation from './StepNavigation';

describe('StepNavigation Component', () => {
  test('should render correctly with default texts', () => {
    const { container } = render(
      <StepNavigation prevHandler={jest.fn} nextHandler={jest.fn} />
    );

    expect(container).toMatchSnapshot();
  });

  test('should render correctly with custom texts', () => {
    const { container } = render(
      <StepNavigation
        prevText="Prev test"
        nextText="Next test"
        prevHandler={jest.fn}
        nextHandler={jest.fn}
      />
    );

    expect(container).toMatchSnapshot();
  });

  test('should render correctly with custom next icon', () => {
    const { container } = render(
      <StepNavigation
        nextIcon="cog"
        prevHandler={jest.fn}
        nextHandler={jest.fn}
      />
    );

    expect(container).toMatchSnapshot();
  });

  test('should call the functions by clicking the buttons', async () => {
    const prevHandler = jest.fn();
    const nextHandler = jest.fn();

    const { getByTestId } = render(
      <StepNavigation prevHandler={prevHandler} nextHandler={nextHandler} />
    );

    const prevButton = getByTestId('prevButton');
    const nextButton = getByTestId('nextButton');

    await fireEvent.click(prevButton);
    await fireEvent.click(nextButton);

    expect(prevHandler).toBeCalledTimes(1);
    expect(nextHandler).toBeCalledTimes(1);
  });
});
