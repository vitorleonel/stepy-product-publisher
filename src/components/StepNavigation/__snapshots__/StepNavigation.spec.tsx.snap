// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepNavigation Component should render correctly with custom next icon 1`] = `
<div>
  <div
    class="steps-item__navigation"
  >
    <button
      class="bp3-button bp3-large"
      data-testid="prevButton"
      type="button"
    >
      <span
        class="bp3-icon bp3-icon-arrow-left"
        icon="arrow-left"
      >
        <svg
          data-icon="arrow-left"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            arrow-left
          </desc>
          <path
            d="M13.99 6.99H4.41L7.7 3.7a1.003 1.003 0 00-1.42-1.42l-5 5a1.014 1.014 0 000 1.42l5 5a1.003 1.003 0 001.42-1.42L4.41 8.99H14c.55 0 1-.45 1-1s-.46-1-1.01-1z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <span
        class="bp3-button-text"
      >
        Previous
      </span>
    </button>
    <button
      class="bp3-button bp3-large bp3-intent-success"
      data-testid="nextButton"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        Next
      </span>
      <span
        class="bp3-icon bp3-icon-cog"
        icon="cog"
      >
        <svg
          data-icon="cog"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            cog
          </desc>
          <path
            d="M15.19 6.39h-1.85c-.11-.37-.27-.71-.45-1.04l1.36-1.36c.31-.31.31-.82 0-1.13l-1.13-1.13a.803.803 0 00-1.13 0l-1.36 1.36c-.33-.17-.67-.33-1.04-.44V.79c0-.44-.36-.8-.8-.8h-1.6c-.44 0-.8.36-.8.8v1.86c-.39.12-.75.28-1.1.47l-1.3-1.3c-.3-.3-.79-.3-1.09 0L1.82 2.91c-.3.3-.3.79 0 1.09l1.3 1.3c-.2.34-.36.7-.48 1.09H.79c-.44 0-.8.36-.8.8v1.6c0 .44.36.8.8.8h1.85c.11.37.27.71.45 1.04l-1.36 1.36c-.31.31-.31.82 0 1.13l1.13 1.13c.31.31.82.31 1.13 0l1.36-1.36c.33.18.67.33 1.04.44v1.86c0 .44.36.8.8.8h1.6c.44 0 .8-.36.8-.8v-1.86c.39-.12.75-.28 1.1-.47l1.3 1.3c.3.3.79.3 1.09 0l1.09-1.09c.3-.3.3-.79 0-1.09l-1.3-1.3c.19-.35.36-.71.48-1.1h1.85c.44 0 .8-.36.8-.8v-1.6a.816.816 0 00-.81-.79zm-7.2 4.6c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`StepNavigation Component should render correctly with custom texts 1`] = `
<div>
  <div
    class="steps-item__navigation"
  >
    <button
      class="bp3-button bp3-large"
      data-testid="prevButton"
      type="button"
    >
      <span
        class="bp3-icon bp3-icon-arrow-left"
        icon="arrow-left"
      >
        <svg
          data-icon="arrow-left"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            arrow-left
          </desc>
          <path
            d="M13.99 6.99H4.41L7.7 3.7a1.003 1.003 0 00-1.42-1.42l-5 5a1.014 1.014 0 000 1.42l5 5a1.003 1.003 0 001.42-1.42L4.41 8.99H14c.55 0 1-.45 1-1s-.46-1-1.01-1z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <span
        class="bp3-button-text"
      >
        Prev test
      </span>
    </button>
    <button
      class="bp3-button bp3-large bp3-intent-success"
      data-testid="nextButton"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        Next test
      </span>
      <span
        class="bp3-icon bp3-icon-arrow-right"
        icon="arrow-right"
      >
        <svg
          data-icon="arrow-right"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            arrow-right
          </desc>
          <path
            d="M14.7 7.29l-5-5a.965.965 0 00-.71-.3 1.003 1.003 0 00-.71 1.71l3.29 3.29H1.99c-.55 0-1 .45-1 1s.45 1 1 1h9.59l-3.29 3.29a1.003 1.003 0 001.42 1.42l5-5c.18-.18.29-.43.29-.71s-.12-.52-.3-.7z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`StepNavigation Component should render correctly with default texts 1`] = `
<div>
  <div
    class="steps-item__navigation"
  >
    <button
      class="bp3-button bp3-large"
      data-testid="prevButton"
      type="button"
    >
      <span
        class="bp3-icon bp3-icon-arrow-left"
        icon="arrow-left"
      >
        <svg
          data-icon="arrow-left"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            arrow-left
          </desc>
          <path
            d="M13.99 6.99H4.41L7.7 3.7a1.003 1.003 0 00-1.42-1.42l-5 5a1.014 1.014 0 000 1.42l5 5a1.003 1.003 0 001.42-1.42L4.41 8.99H14c.55 0 1-.45 1-1s-.46-1-1.01-1z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <span
        class="bp3-button-text"
      >
        Previous
      </span>
    </button>
    <button
      class="bp3-button bp3-large bp3-intent-success"
      data-testid="nextButton"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        Next
      </span>
      <span
        class="bp3-icon bp3-icon-arrow-right"
        icon="arrow-right"
      >
        <svg
          data-icon="arrow-right"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            arrow-right
          </desc>
          <path
            d="M14.7 7.29l-5-5a.965.965 0 00-.71-.3 1.003 1.003 0 00-.71 1.71l3.29 3.29H1.99c-.55 0-1 .45-1 1s.45 1 1 1h9.59l-3.29 3.29a1.003 1.003 0 001.42 1.42l5-5c.18-.18.29-.43.29-.71s-.12-.52-.3-.7z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
